lib/puppet/application/master.rb

```diff
@@ -172,7 +172,7 @@ def compile
         raise "Could not compile catalog for #{options[:node]}"
       end
 
-      puts PSON::pretty_generate(catalog.to_resource, :allow_nan => true, :max_nesting => false)
+      puts JSON::pretty_generate(catalog.to_resource, :allow_nan => true, :max_nesting => false)
```

Comment:
I think it works because the catalog object has a `to_json` method on it that it gets from `Puppet::Network::FormatSupport`